### PR TITLE
Amy finish focus gameObject

### PR DESCRIPTION
**Pull request by AmyOrz**

### CODE REVIEW

File: test/integration/job/initHotKeysJob_test.re [hunk 1]
```diff
@@ -139,12 +144,101 @@ let _ =
...
+          MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode();
+
           triggerDeleteHotKeyEvent();
 
           BuildComponentTool.buildSceneTree(TestTool.buildEmptyAppState())
           |> ReactTestTool.createSnapshotAndMatch;
         })
       );
+
+      describe("test bind focus hot-key", () => {
```
Comment (yyc-git) [inline test/integration/job/initHotKeysJob_test.re]: TODO test: should test "set distance"

File: test/integration/job/initHotKeysJob_test.re [hunk 2]
```diff
@@ -139,12 +144,101 @@ let _ =
...
+
+              editorState
+              |> SceneViewEditorService.unsafeGetEditCamera
+              |. GameObjectComponentEngineService.unsafeGetArcballCameraControllerComponent(
+                   engineState,
+                 )
+              |. ArcballCameraEngineService.unsafeGetArcballCameraControllerTarget(
+                   engineState,
+                 )
+              |> expect == (0., 0., 0.);
```
Comment (yyc-git) [inline test/integration/job/initHotKeysJob_test.re]: TODO test: should  first set target to (1.0,2.0,3.0); then focus scene;

File: test/integration/job/initHotKeysJob_test.re [hunk 3]
```diff
@@ -139,12 +144,101 @@ let _ =
...
+                |> expect == targetPosition;
+              },
+            )
+          );
+        });
+
+        describe("else", () =>
+          test(
+            "key down f, arcballCamera should set target to be (0.,0.,0.)", () => {
+            let editorState = StateEditorService.getState();
```
Comment (yyc-git) [inline test/integration/job/initHotKeysJob_test.re]: TODO test: should:
1.clear current scene tree node;
2.set target to (1.0,2.0,3.0);
3.focus

File: src/service/stateTuple/logic/ArcballCameraControllerLogicService.re [hunk 4]
```diff
@@ -185,4 +185,60 @@ let unbindGameViewActiveCameraArcballCameraControllerEvent = engineState =>
...
+  );
+
+  let editorCameraArcballControllerComponent =
+    editorState
+    |> SceneViewEditorService.unsafeGetEditCamera
+    |. GameObjectComponentEngineService.unsafeGetArcballCameraControllerComponent(
+         engineState,
+       );
+
+  isCurrentSceneTreeNodeCanBeOperate ?
```
Comment (yyc-git) [inline src/service/stateTuple/logic/ArcballCameraControllerLogicService.re]: TODO refactor: should judge "whether current scene tree node is scene"

File: src/service/stateTuple/logic/ArcballCameraControllerLogicService.re [hunk 5]
```diff
@@ -185,4 +185,60 @@ let unbindGameViewActiveCameraArcballCameraControllerEvent = engineState =>
...
+              engineState,
+            )
+         |. TransformEngineService.getPosition(engineState),
+       )
+    |> ArcballCameraEngineService.setArcballCameraControllerDistance(
+         distance,
+         editorCameraArcballControllerComponent,
+       ) :
+    engineState
+    |> ArcballCameraEngineService.setArcballCameraControllerTarget(
```
Comment (yyc-git) [inline src/service/stateTuple/logic/ArcballCameraControllerLogicService.re]: TODO feat: should set distance

Comment (AmyOrz) [reply src/service/stateTuple/logic/ArcballCameraControllerLogicService.re]: if currentGameObject is scene or don't have it, we don't need set distance

File: src/service/stateTuple/logic/ArcballCameraControllerLogicService.re (hunk 4 above)
Comment (AmyOrz) [reply src/service/stateTuple/logic/ArcballCameraControllerLogicService.re]: there is current scene tree node is scene or don't have it

Comment (yyc-git) [reply src/service/stateTuple/logic/ArcballCameraControllerLogicService.re]: should judge explictly

**Comment by yyc-git** [comment]
TODO feat: if has no current scene tree node, do nothing

**Comment by yyc-git** [comment]
TODO fix: split isCurrentSceneTreeNodeCanBeOperate to isCurrentSceneTreeNodeSceneGameObject, isCurrentSceneTreeNodeSceneChildren

File: src/service/stateTuple/logic/ArcballCameraControllerLogicService.re [hunk 6]
```diff
@@ -187,8 +187,25 @@ let unbindGameViewActiveCameraArcballCameraControllerEvent = engineState =>
        StateEditorService.getStateIsDebug(),
      );
 
+let _getGameObjectPosition = (gameObject, engineState) =>
+  engineState
+  |> GameObjectComponentEngineService.unsafeGetTransformComponent(gameObject)
+  |. TransformEngineService.getPosition(engineState);
+
+let _setArcballCameraControllerAttribute =
```
Comment (yyc-git) [inline src/service/stateTuple/logic/ArcballCameraControllerLogicService.re]: TODO refactor: rename to _setArcballCameraControllerDistanceAndTarget

File: src/service/stateTuple/logic/ArcballCameraControllerLogicService.re [hunk 7]
```diff
@@ -187,8 +187,25 @@ let unbindGameViewActiveCameraArcballCameraControllerEvent = engineState =>
...
+    (arcballCameraController, (distance, position), engineState) =>
```
Comment (yyc-git) [inline src/service/stateTuple/logic/ArcballCameraControllerLogicService.re]: TODO refactor: rename position to target

**Pull request merged**